### PR TITLE
Improves unit test code coverage for azure_managedDiskController.go

### PR DESCRIPTION
**Pull request by weijiehu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:
Improves unit test code coverage for azure_managedDiskController.go

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig cloud-provider
/area provider/azure
/priority important-soon

### CODE REVIEW

**Review by nilo19 — COMMENTED** [review]
> /retest

no need to retest, just fix the linting problem and force push your commit

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_managedDiskController_test.go [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+	defer ctrl.Finish()
+
+	maxShare := int32(2)
+	goodDiskEncryptionSetID := fmt.Sprintf("/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/diskEncryptionSets/%s", "diskEncryptionSet-name")
+	badDiskEncryptionSetID := fmt.Sprintf("badDiskEncryptionSetID")
+	testTags := make(map[string]*string)
+	testTags[WriteAcceleratorEnabled] = to.StringPtr("true")
+	testCases := []struct {
+		desc                string
+		diskId              string
```
[the remark this answers is not in the document]
Comment (weijiehu) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_managedDiskController_test.go:47]: ok, will fix it.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by weijiehu** [comment]
@feiskyer I update the PR with fix Zones nil case, needs lgtm label, thanks.

**Review by andyzhangx — APPROVED** [review]
/approve

**Pull request merged**